sound/soc/intel/boards/sof_sdw_rt_amp.c

```diff
@@ -196,26 +196,28 @@ int rt_amp_spk_rtd_init(struct snd_soc_pcm_runtime *rtd)
 
 	rt_amp_map = get_codec_name_and_route(rtd, codec_name);
 
+	if (!strstr(card->components, "spk:")) {
```

Comment:
right, but we still have a confusion between speaker and amp. What you want is a list of amps, not a list of speakers.

Reply:
> right, but we still have a confusion between speaker and amp. What you want is a list of amps, not a list of speakers.

Good point, but IMHO changing `spk:` tag to `amp:` is a different topic. And it will break the new kernel + old UCM case.